spring-xd-dirt/src/main/java/org/springframework/xd/dirt/plugins/stream/ModuleTypeConversionPlugin.java

```diff
@@ -66,15 +70,26 @@ public ModuleTypeConversionPlugin(Collection<AbstractFromMessageConverter> conve
 		this.converterFactory = new CompositeMessageConverterFactory(converters);
 	}
 
+	/**
+	 * Get the underlying composite message converter factory which could be further used by any other
+	 * plugin that requires to apply module type conversion explicitly.
+	 * See {@link org.springframework.xd.dirt.plugins.spark.streaming.SparkStreamingMessageConverterSupport}
+	 *
+	 * @return return the composite message converter factory
+	 */
+	public CompositeMessageConverterFactory getConverterFactory() {
+		return this.converterFactory;
+	}
+
 	@Override
 	public void postProcessModule(Module module) {
 		String outputType = null;
 		String inputType = null;
 		if (module.getType() == ModuleType.source || module.getType() == ModuleType.processor) {
-			outputType = module.getProperties().getProperty("outputType");
+			outputType = module.getProperties().getProperty(OUTPUT_TYPE);
```

Comment:
Ideally, we should avoid using `getProperties()`, so the code that does this should be factored out. getMimeType() in SparkStreamingPlugin is exactly what we want, so should convert it to utility code.